src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -253,6 +260,14 @@ impl<'a> RLookupKey<'a> {
         }
     }
 
+    /// Returns the name of this key as a `&CStr`.
+    ///
+    /// This is used internally (load_document code paths) to access a reference with the correct lifetime.
+    pub(crate) fn name_ref(&self) -> &'a CStr {
+        // Safety: We assume the pointer is valid and points to a null-terminated C string.
+        unsafe { CStr::from_ptr(self.name) }
+    }
+
```

Comment:
don't use the `name` C ptr here, that only exists for backwards compatibility with C. Use the `_name` field and return a `Cow<'a, CStr>` from this method